libbpf-tools/llcstat.c

```diff
@@ -143,11 +150,20 @@ static void print_map(struct bpf_map *map)
 			return;
 		}
 		hit = info.ref > info.miss ? info.ref - info.miss : 0;
-		pid = next_key >> 32;
-		cpu = next_key;
-		printf("%-8u %-16s %-4u %12llu %12llu %6.2f%%\n", pid, info.comm,
-			cpu, info.ref, info.miss, info.ref > 0 ?
-			hit * 1.0 / info.ref * 100 : 0);
+		cpu = next_key.cpu;
+		pid = next_key.pid;
+		tid = next_key.tid;
+		if (env.per_thread) {
+			printf("%-8u %-8u %-16s %-4u %12llu %12llu %6.2f%%\n",
```

Comment:
similar to the other comment in this review pass, it should be possible to build up this formatted output more gradually such that the if statement only does tid-specific work